File: implementations/flutter/design_patterns_flutter/lib/patterns/creational/builder/builder.dart
import 'package:flutter/widgets.dart';
import 'product/card.dart';

abstract class CardBuilder {
  CardBuilder setTitle(String title);
  CardBuilder setSubtitle(String subtitle);
  CardBuilder setMedia(String url);
  CardBuilder setBody(Widget content);
  CardBuilder setFooter(Widget content);
  CardProduct build();
}

File: implementations/react-native/design-patterns-rn/src/patterns/creational/builder/builder.ts
import type { ReactNode } from 'react';
import { CardProduct } from './product-card';

export interface CardBuilder {
  setTitle(title: string): this;
  setSubtitle(subtitle: string): this;
  setMedia(url: string): this;
  setBody(content: ReactNode): this;
  setFooter(content: ReactNode): this;
  build(): CardProduct;
}
